### PR TITLE
E2e test suite from Nodejs worker

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
The goal here was to use the same [test suite as Nodejs](https://github.com/Azure/azure-functions-nodejs-worker/tree/dev/test/end-to-end) so that we can later refactor to a separate project used by _all workers_.

This does just that - with ZERO functional changes to the tests (only string changes for different resource names)

These tests work against actual Azure resources.

I figured folks wouldn't want to run e2e locally, so they will only run in CI

resolves https://github.com/Azure/azure-functions-powershell-worker/issues/75

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
cc @anirudhgarg @paulbatum @fabiocav for awareness

File: test/E2E/TestFunctionApp/extensions.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (pragnagopa) [inline test/E2E/TestFunctionApp/extensions.csproj]: Do not check in this file. Use `func extensions install` command. This command will generate extensions.csproj with the latest nuget packages.

Comment (TylerLeonhardt) [reply test/E2E/TestFunctionApp/extensions.csproj]: How does it get generated? Does it look at the bindings?

Comment (pragnagopa) [reply test/E2E/TestFunctionApp/extensions.csproj]: func cli parses function.json and generates extensions.csproj

**Comment by TylerLeonhardt** [comment]
I'm going to merge these in. Feel free to give me feedback when you get back @daxian-dbw!

**Pull request merged**